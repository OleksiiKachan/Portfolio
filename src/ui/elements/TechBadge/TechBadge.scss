@import '../../assets/style/variables';
@import '../../assets/style/mixins';

.techBadge {
  border-radius: 10px;
  padding: 8px 15px;
  font-size: 0.75rem;
  font-weight: 600;
  box-shadow: 2px 4px 4px rgba(0, 0, 0, 0.25);

  &_dark {
    background-color: $color-grey;
    color: $color-white;
  }

  &_light {
    background-color: $color-white;
    color: $color-grey;
  }
}
