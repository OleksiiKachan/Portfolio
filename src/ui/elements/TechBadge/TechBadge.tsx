import React, { ReactNode } from 'react';
import classNames from 'classnames';
import './TechBadge.scss';

/*
    <TechBadge>{''}</TechBadge>
*/

type PropsType = {
  children: ReactNode;
  type: string;
  className?: string;
};

export default ({ children, type, className }: PropsType) => {
  return (
    <span className={classNames('techBadge', `techBadge_${type}`, className)}>
      {children}
    </span>
  );
};
