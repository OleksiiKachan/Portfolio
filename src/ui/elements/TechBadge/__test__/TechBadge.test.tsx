import React from 'react';
import { shallow } from 'enzyme';
import TechBadge from '../TechBadge';

describe('TechBadge', () => {
  it('Snapshot', () => {
    const component = shallow(<TechBadge>Tech Badge</TechBadge>);

    expect(component).toMatchSnapshot();
  });
});
