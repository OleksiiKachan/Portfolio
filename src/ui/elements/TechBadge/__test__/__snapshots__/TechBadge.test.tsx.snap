// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechBadge Snapshot 1`] = `
<span
  className="techBadge techBadge_undefined"
>
  Tech Badge
</span>
`;
